### ISSUE TITLE
Double coalescing confuses `explode` analyser

### ISSUE THREAD

**Issue by someniatko** [vimeo/psalm#9174]
https://psalm.dev/r/7fb976f0f7

Should report no issue in both functions.

**Comment by Boldewyn** [comment]
I’ve been biten by this bug, too. Upgrading from 5.4.0 to 5.6.0 left me with a `PossiblyUndefinedArrayOffset` for code that looks like this:

```php
<?php

function test(string $input) : void {
    if (strpos($input, '=') === false) {
        return;
    }
    list($key, $value) = explode('=', $input, 2); # ! here I know that explode returns
                                                  # an array of length 2, but Psalm reports an error
}

test('a=b');
```

**Comment by weirdan** [comment]
@Boldewyn that's beyond Psalm's understanding, so you need to help it a bit: https://psalm.dev/r/3140be0736

It doesn't seem to have anything to do with the original report though.

**Comment by Boldewyn** [comment]
OK, understood! Thank you!

It just happens that psalm started to report these lines in versions > 5.4.0, and the snippet in the original post seemed to match somewhat the case I saw.

**Comment by someniatko** [comment]
@Boldewyn It seems the thing which is introduced in 5.5.0 is not the bug itself, but rather the new feature - better types of `explode` function, as done in https://github.com/vimeo/psalm/pull/9016

And that feature exposed the bug I'm reporting here. And this is also the reason why you too get the error.

**Comment by someniatko** [comment]
I have updated the issue title and first message description. Provided there the reproducer which does not rely on `explode` types.

**Comment by orklah** [comment]
Hum, weird, doc says that null coalescing should suppress the error: https://psalm.dev/docs/running_psalm/issues/PossiblyUndefinedArrayOffset/

https://psalm.dev/r/97e3b85fac (I posted your snippet again for better triage)
